ui: [depends on #42817] Links on decommissioned node history page

koorosh opened the pull request:
Navigation to Decommissioned node history page has to be possible from two places:
- Nodes Overview page -> Decommissioned nodes table
- Debug page -> Node Diagnostics section

koorosh committed:
ui: Add route to nodes history page

It has to be discussed about appropriate path name.

Release note: None

koorosh committed:
ui: Add Decommissioned nodes history page

- url path for new page is following: '#/reports/nodes/history',
please let me know if it has to be changed to something more meaningful.
I decided to make it more generic in case we want to extend it with
other 'historical' content. Also, it is nested route under 'nodes' as it
belongs to this entity.
- this page is supposed to display the history of all decommissioned
nodes in one place;
- the logic of retrieving and displaying nodes are almost the same as
in Nodes Overview container with adjustments according to provided
designs;
- Redux selectors, and helper functions are refactored to external
utils module so it reduces code duplication;

Release note (admin ui change): Decommissioned node history page is added as a dedicated page to reduce amount of information on main Cluster Overview page (issue: #24636)

koorosh committed:
ui: Fix import paths in DecommissionedNodeHistory container

Release note: None

koorosh committed:
ui: Add 'Decommissioned node history' link on Nodes Overview page

Release note (admin ui change): Link on Decommissioned Node List from Cluster Overview page (#24636)

koorosh committed:
ui: Add 'Decommissioned node history' link on Debug page

Release note (admin ui change): Link on Decommissioned Node List from Debug page (#24636)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42820)
<!-- Reviewable:end -->

koorosh committed:
ui: Limit the number of most recent decommissioned nodes to display

- Display only 5 most recent decommissioned nodes on Cluster Overview view;
- Provide a link to navigate to Decommissioned Node History page;

Release note: None

koorosh committed:
ui: Display absolute time for decommissioned nodes

- Represent decommissioning date in absolute format instead of relative.

Release note: None

koorosh committed:
ui: Fix typo in a comment

Release note: None

koorosh committed:
ui: Simplify imports for "reports" components

- Main router component refers to most Reports containers and their imports weren't
aggregated into single import entry point. To reduce number of import statements, all
reports containers are re-imported from top level index file.
- 'nodesHistory' directory is renamed to 'nodeHistory'

Release note: None

knz commented:
I'm not sure what happened but it seems that you based your changes off the wrong branch.

Koorosh commented:
Current changes are irrelevant with the latest changes for the Nodes List view (branch: ui-node-list-view-redesign)
